tensorflow/contrib/seq2seq/python/ops/layers.py

```diff
@@ -20,15 +20,185 @@
 from __future__ import division
 from __future__ import print_function
 
+from tensorflow.python.ops import tensor_array_ops
 from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import variable_scope as vs
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import control_flow_ops
+from tensorflow.python.ops import rnn
 
+from tensorflow.contrib.layers import fully_connected
+
+from functools import partial
 
 __all__ = ["rnn_decoder",
            "rnn_decoder_attention"]
 
+"""Used to project encoder state in `rnn_decoder`"""
+encoder_projection = partial(fully_connected, activation_fn=math_ops.tanh)
```

Comment:
ah i see what you do here.  actually, i prefer you have the argument to encoder_projection be None, and inside the code do:

if encoder_projection is None:
  encoder_projection = partial(...)

keep it out of the global namespace.